### PR TITLE
Client module

### PR DESCRIPTION
**Pull request by vicennial**

### CODE REVIEW

File: connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/SparkSession.scala [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  private[client] def newDataset(f: proto.Relation.Builder => Unit): Dataset = {
+    val builder = proto.Relation.newBuilder()
+    f(builder)
+    val plan = proto.Plan.newBuilder().setRoot(builder).build()
+    new Dataset(this, plan)
+  }
+
+  private[client] def analyze(plan: proto.Plan): proto.AnalyzePlanResponse = {
+    // TODO: Add dependencies check once implemented
+    // dependencies.ensureAllDependenciesUploaded()
```
Comment (hvanhovell) [inline connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/SparkSession.scala]: Please remove this comment.

File: connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/Dataset.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (hvanhovell) [inline connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/Dataset.scala]: The location should be ` connector/connect/client/src/main/scala/org/apache/spark/sql/connect/client/Dataset.scala` and not ` connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/Dataset.scala`

File: connector/connect/client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        <sbt.project.name>connect-client</sbt.project.name>
+        <guava.version>31.0.1-jre</guava.version>
+        <guava.failureaccess.version>1.0.1</guava.failureaccess.version>
+        <io.grpc.version>1.47.0</io.grpc.version>
+        <tomcat.annotations.api.version>6.0.53</tomcat.annotations.api.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
```
Comment (hvanhovell) [inline connector/connect/client/pom.xml]: Please add a comment that states that this is a temporary measure.

File: connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/Dataset.scala (hunk 2 above)
Comment (vicennial) [reply connector/connect/client/src/main/scala/org.apache.spark.sql.connect.client/Dataset.scala]: Argh, thought Intellij would automatically expand here

File: connector/connect/client/src/main/scala/org/apache/spark/sql/connect/client/Dataset.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.apache.spark.sql.Row
+
+class Dataset(val session: SparkSession, private[client] val plan: proto.Plan) {
+
+  /**
+   * Placeholder method.
+   * @return
+   */
+  def collect(): Array[Row] = {
+    new Array[Row](1)
```
Comment (hvanhovell) [inline connector/connect/client/src/main/scala/org/apache/spark/sql/connect/client/Dataset.scala]: Nit: Return an empty array? Now you would get null if you try to access this.

**Review by hvanhovell — APPROVED** [review]
LGTM - can you open a PR against apache/spark?

**Comment by vicennial** [comment]
@hvanhovell https://github.com/apache/spark/pull/39078

**Pull request closed without merging**